resources/api/reservation.py

```diff
@@ -134,29 +134,40 @@ def __init__(self, *args, **kwargs):
         # set supported and required extra fields
         if resource:
             cache = self.context.get("reservation_metadata_set_cache")
-            supported = resource.get_supported_reservation_extra_field_names(
-                cache=cache
-            )
-            required = resource.get_required_reservation_extra_field_names(cache=cache)
-
             # staff events have less requirements
             request_user = self.context["request"].user
             is_staff_event = data.get("staff_event", False)
 
+            # staff events always have the same set of required fields
             if is_staff_event and resource.can_create_staff_event(request_user):
                 required = {"reserver_name", "event_description"}
 
+                # billing fields are also required if own use
+                if data.get("type", None) == Reservation.TYPE_NORMAL:
+                    required |= {
```

Comment:
Just to make sure I got this right, these are only required if the fields are in the reservation metadata set? So for free resources that have a reservation form (metadata set) without these fields, they are also not required, right?